src/nimvscode/nimSuggestExec.nim

```diff
@@ -29,7 +29,7 @@ type NimSuggestProcessDescription* = ref object
 
 var nimSuggestPath: cstring
 var nimSuggestVersion: cstring
-var nimSuggestProcessCache = newJsAssoc[cstring, Future[
+var nimSuggestProcessCache = newMap[cstring, Future[
```

Comment:
This is better, but I'm wondering if there is an issue on line 157 with the jsDelete.

Reply:
It removes the key from the process map right?
As far as i understand, this should only be done if the closeCachedProcess is sucessful. is this what you're refering to?

Reply:
`jsDelete` is equivalent to the `delete` keyword in JS.

There is a `delete` proc on `Map` itself, checkout `jsNode.nim`

Reply:
considering its deleting only the entry ( which was just closed and undefined ) , i don't think it matters.

Reply:
This: https://nim-lang.org/docs/jsffi.html#jsDelete%2Cauto

Which is this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete

Is not this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete

Reply:
There are some JS shenanigans going on here, so if you're not familiar that's ok.

The gist is that one is an old school operator and Map which is a newer API, has a method on it.

Reply:
thats fine, i'll be changing the implementation here